web/src/main/java/com/synopsys/integration/alert/web/api/job/JobConfigActions.java

```diff
@@ -324,33 +325,56 @@ protected ValidationActionResponse validateWithoutChecks(JobFieldModel resource)
+                if (optionalChannelDistributionTestAction.isPresent()) {
+                    ChannelDistributionTestAction channelDistributionTestAction = optionalChannelDistributionTestAction.get();
                     Map<String, ConfigurationFieldModel> fields = createFieldsMap(channelFieldModel, otherJobModels);
                     // The custom message fields are not written to the database or defined fields in the database.  Need to manually add them.
                     // TODO Create a mechanism to create the field accessor with a combination of fields in the database and fields that are not.
                     Optional<ConfigurationFieldModel> topicField = convertFieldToConfigurationField(channelFieldModel, TestAction.KEY_CUSTOM_TOPIC);
                     Optional<ConfigurationFieldModel> messageField = convertFieldToConfigurationField(channelFieldModel, TestAction.KEY_CUSTOM_MESSAGE);
+                    Optional<ConfigurationFieldModel> destinationField = convertFieldToConfigurationField(channelFieldModel, TestAction.KEY_DESTINATION_NAME);
                     topicField.ifPresent(model -> fields.put(TestAction.KEY_CUSTOM_TOPIC, model));
                     messageField.ifPresent(model -> fields.put(TestAction.KEY_CUSTOM_MESSAGE, model));
-                    TestAction testAction = testActionOptional.get();
-                    FieldUtility fieldUtility = new FieldUtility(fields);
-                    String jobId = channelFieldModel.getId();
+                    destinationField.ifPresent(model -> fields.put(TestAction.KEY_DESTINATION_NAME, model));
 
-                    MessageResult providerTestResult = testProviderConfig(fieldUtility, jobId, channelFieldModel);
+                    MessageResult providerTestResult = testProviderConfig(new FieldUtility(fields), jobIdString, channelFieldModel);
                     if (providerTestResult.hasErrors()) {
                         responseModel = ValidationResponseModel.fromStatusCollection(providerTestResult.getStatusMessage(), providerTestResult.getFieldStatuses());
                         return new ValidationActionResponse(HttpStatus.OK, responseModel);
                     }
 
-                    MessageResult testActionResult = testAction.testConfig(jobId, channelFieldModel, fieldUtility);
+                    // Not all channels have a global config
+                    ConfigurationModel nullableChannelGlobalConfig = configurationAccessor.getConfigurationsByDescriptorNameAndContext(channelFieldModel.getDescriptorName(), ConfigContextEnum.GLOBAL)
+                                                                         .stream()
+                                                                         .findFirst()
+                                                                         .orElse(null);
+
+                    List<BlackDuckProjectDetailsModel> projectFilterDetails = Optional.ofNullable(resource.getConfiguredProviderProjects()).orElse(List.of())
```

Comment:
Resource should return an empty list instead of null so we can process without checking for anything.

Reply:
`resource` is a REST API request-body, so this can absolutely be null.